### DISCUSSION TITLE
Local images not shown

### DISCUSSION THREAD

**Discussion by apperside** [tamagui/tamagui#336] [General]
**Describe the bug**
I am trying to integrate cross platform images in my tamagui app.
I started from create-tamagui-app, and now I am trying to add an `Image` component, so I added an `assets` folder to my app

![Screenshot 2022-09-19 at 12 32 52](https://user-images.githubusercontent.com/5955338/190999252-2c80a963-c71e-4b72-8662-e5cfdd65d1e9.png)

Then inside my packages/app/features/home/screen.ts file I am trying to require it

```typescript
const logo = require('../../assets/logo.png')`
```

and to render it with

```typescript
<Image  style={{ width: 100, height: 100 }} source={{ uri: logo }}></Image>
```

or

```typescript
<Image  style={{ width: 100, height: 100 }} source={{ logo }}></Image>
```
But there is no way to make it show up.

I also tried to import the `Image` component directly from react-native instead of `tamagui` without luck.

What is the proper way to render local assets?

Thank you

**To Reproduce**
Try to render a local image

**System Info (please complete the following information):** 

  System:
    OS: macOS 12.6
    CPU: (16) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 22.40 MB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.17.0 - /usr/local/bin/node
    Yarn: 3.2.0 - ~/.nvm/versions/node/v16.17.0/bin/yarn
    npm: 8.15.0 - ~/.nvm/versions/node/v16.17.0/bin/npm
  Browsers:
    Chrome: 105.0.5195.125
    Firefox: 104.0.2
    Safari: 16.0

**Comment by hakkokimkr** [comment]
``` <Image style={{ width: 100, height: 100 }} source={ logo }></Image> ```

Remove parenthesis around logo like above

**Comment by apperside** [comment]
Sorry @dividenconquer , it was just a copy/paste mistake, what you said is what I did

**Comment by apperside** [comment]
After further investigation I think it is a assets resolution problem:

if inside the native app, I return a component which renders an image located in the native project (eg: /apps/expo/src/assets) it works

if inside the shared project directory and add here an asset file, it doesn't work


![Screenshot 2022-09-19 at 16 49 09](https://user-images.githubusercontent.com/5955338/191047751-edde049a-494b-416e-bf3d-ff3079457564.png)

![Screenshot 2022-09-19 at 16 52 51](https://user-images.githubusercontent.com/5955338/191047789-1c3c4360-0d0e-48b7-adee-0c2d44ac9b32.png)

**Comment by apperside** [comment]
In addition it doesn't work neither in the next js project

**Comment by apperside** [comment]
Hello @natew ,
I just installed the beta 177, there a lot of improvements and fixes, but the Image component keep working bad.
It works now on react native side, but the shared screen on the web doesn't show the image. 
Could it be related to where assets are stored? where shared assets should be put in an app created by create-tamagui-app?

**Comment by apperside** [comment]
I also just tried from a fresh app created with create-tamagui-app, on the web there is no way to make the image show up.
[Here](https://github.com/apperside/tamagui-image-bug) you can find a repository with the bug reproduction, it's just a fresh tamagui app with an image added in /packages/app/features/home/screen.tsx

**Comment by apperside** [comment]
Hello @natew , any update on this? 

At the moment, I ended up creating an `Image.web.tsx` and `Image.native.tsx` components, rendering `next/image` on web and  `Image` from `tamagui` on mobile. 

There are some problems related to how next-js and react native handles images sizing, but it works for now. 

In the attempt to understand what is causing the problem, I went through the source code and I found that your `Image` component just renders the `Image` component from `react-native`, but I'm not sure it is gonna work: if I use the `react-native`'s `Image` component directly in the next-js app it doesn't work, so I was wondering if the actual implementation was supposed to work on both platforms.

**Comment by mustachebutter** [comment]
Hey! Same problem with me. I tried putting my image in a folder inside `packages/app/features/assets`. It only renders the image on mobile. I have tested this with both `Image` components from 'tamagui' and 'react-native'. If the source is from the web then RN and Next app load the image normally.

**Comment by natew** [comment]
@apperside it goes to react-native-web which handles the image.

I think this is an issue in the starter perhaps react native only allows images from within expo directory? If so the fix is just with the monorepo, you could maybe just move the `app` directory directly inside of `expo` and rename `expo` to `app` or something creative. Or perhaps there's some metro plugin or something.

But moving to discussion as the monorepo template could have infinite development, and we'd need a separate section or something for all those. Would happily accept a PR that works around this.

**Comment by nandorojo** [comment]
This may be related to an outdated issue with the Solito starter. 

Please make sure to do the following:

1. Add `next-images` to your `next.config.js`
2. Add `images.disabledStaticImages = true` to your `next.config.js`

See the updated Solito starter's `next.config.js` here: https://github.com/nandorojo/solito/blob/master/example-monorepos/blank/apps/next/next.config.js

**Reply by natew** [reply]
Gave it a go here f6377e77e

**Comment by diazsasak** [comment]
different but similar issue, i solve it using isWeb to make it work on both.

![image](https://github.com/tamagui/tamagui/assets/7880137/d4ada877-0da8-4898-99fc-4dcfbdc38626)
